### PR TITLE
Math optimizations

### PR DESCRIPTION
**Pull request by smilediver**
Mostly moves some trivial and commonly used methods in math classes to headers so that they can be inlined by compiler.

### CODE REVIEW

**Comment by paulocoutinhox** [comment]
What is inl file?

**Review by halx99 — CHANGES_REQUESTED** [review]
**I prefer just implement inline APIs in class define directly, don't create individual .inl file if the inline APIs are public**

File: core/math/Color.inl
Comment (halx99) [inline core/math/Color.inl:1]: don't create individual .inl file

File: core/math/Mat4.inl
Comment (halx99) [inline core/math/Mat4.inl:1]: don't create individual .inl file

File: core/math/Rect.inl
Comment (halx99) [inline core/math/Rect.inl:1]: don't create individual .inl file

File: core/math/Vec4.inl
Comment (halx99) [inline core/math/Vec4.inl:1]: don't create individual .inl file

File: core/math/Quaternion.inl
Comment (halx99) [inline core/math/Quaternion.inl:1]: move to class define body

**Comment by smilediver** [comment]
Should I put definitions with declarations like so:

```cpp
class Color {
    void method() { doStuff(); }
};
```
or keep declarations clean and put definitions at the end of file like this:
```cpp
class Color {
    void method();
};

void Color::method() { doStuff(); }
```
or maybe something in between, like keep shorter definitions with declarations, but keep larger definitions at the end of file?

**Comment by halx99** [comment]
I prefer first one, less code

**Review by halx99 — APPROVED** [review]

**Pull request merged**